### PR TITLE
fix: copy to Clipboard order

### PR DESCRIPTION
**Pull request by AAfghahi**
### SUMMARY
This fixes this issue: https://github.com/apache/superset/issues/16149

The reason for this issue occurring is that Javascript orders objects based on integer and integer-like keys first, and then all strings: https://2ality.com/2015/10/property-traversal-order-es6.html

Therefore to prep the data for copyToClipboard, this code goes through and orders the keys based on the column names. 

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Also fixed the table order, which was previously wrong. 
Before: 
Screen Shot 2021-08-20 at 10 20 21 AM

After:
Screen Shot 2021-08-19 at 11 39 13 AM

### TESTING INSTRUCTIONS
To test:
1) Create a google sheet/database that has a mix of integers and strings as columns
2) run a query on the database
3) Use the copy to clipboard feature. 
4) Go into explore, use the copy to clipboard feature. 

Also-
1) Go to charts
2) make sure the different visualization types have the south data pane rendering 

### ADDITIONAL INFORMATION

- [ x] Has associated issue:  https://github.com/apache/superset/issues/16149
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx [hunk 1]
```diff
@@ -316,7 +315,10 @@ export const DataTablesPane = ({
   const TableControls = (
     <TableControlsWrapper>
       <RowCount data={data[activeTabKey]} loading={isLoading[activeTabKey]} />
-      <CopyToClipboardButton data={data[activeTabKey]} />
+      <CopyToClipboardButton
+        data={data[activeTabKey]}
+        columns={queryFormData.all_columns}
```
Comment (hughhhh) [inline superset-frontend/src/explore/components/DataTablesPane/index.tsx:320]: nit: deconstruct all_columns

```
const { all_columns } = queryFormData
```

**Review by hughhhh — APPROVED** [review]

File: superset-frontend/src/explore/components/DataTablesPane/index.tsx (hunk 1 above)
Comment (AAfghahi) [reply superset-frontend/src/explore/components/DataTablesPane/index.tsx]: made the changes!

**Comment by jinghua-qa** [comment]
added test cases

**Pull request merged**